### PR TITLE
Better documentation, class hierarchy fixups

### PR DESCRIPTION
**Pull request by voetsjoeba**
- Adds better documentation. As discussed with Mahmoud, the background section on the JKS/JCEKS page can use more fleshing out, but this can already serve for now. My main goal is to provide technical details for using the modules.
 - Introduces a new AbstractKeystore class that both KeyStore and BksKeyStore inherit from. Previously BksKeyStore was a subclass of KeyStore, inheriting JKS-specific properties like .private_keys and .secret_keys that don't make sense in BksKeyStore.
 - Adds a .version attribute on KeyStore and BksKeyStore so users can query the version of the store format that was loaded.

### CODE REVIEW

File: tools/readks.py [hunk 1]
```diff
@@ -68,12 +68,14 @@ def get_entry_bits(entry):
...
-    parser = ArgumentParser()
+    parser = ArgumentParser(description="Utility for reading Java keystores.")
     parser.add_argument("keystore_file")
     parser.add_argument("keystore_password")
-    parser.add_argument("--type", default="jks")
-    parser.add_argument("-l", "--list", action="store_true", default=True)
-    parser.add_argument("-x", "--extract", metavar="ALIAS", dest="extract_alias")
+    parser.add_argument("--type", default="jks", choices=["jks", "jceks", "bks", "uber"], help="The type of input keystore. Defaults to 'jks'.")
+
+    group = parser.add_mutually_exclusive_group()
```
Comment (mahmoud) [inline tools/readks.py]: v nice

**Comment by mahmoud** [comment]
Looks great! Merging, might do some cleanup once I get it published.

**Comment by mahmoud** [comment]
@voetsjoeba take a look: http://pyjks.readthedocs.io/ :)

**Comment by voetsjoeba** [comment]
Cool, thanks :)

**Pull request merged**